### PR TITLE
handle error in GlyphSource#getSimpleGlyphs

### PR DESCRIPTION
**Pull request by stepankuzmin**
Is this a correct approach to #2711? Should we inherit `GlyphSource` from `Evented` to fire errors?

### CODE REVIEW

File: js/symbol/glyph_source.js [hunk 1]
```diff
@@ -82,15 +84,19 @@ class GlyphSource {
...
-            if (!err) {
-                const stack = this.stacks[fontstack][range] = data.stacks[0];
-                for (let i = 0; i < missing[range].length; i++) {
-                    const glyphID = missing[range][i];
-                    const glyph = stack.glyphs[glyphID];
-                    const rect  = atlas.addGlyph(uid, fontstack, glyph, buffer);
-                    if (glyph) glyphs[glyphID] = new SimpleGlyph(glyph, rect, buffer);
-                }
+            if (err) {
+                this.fire('error', {error: err});
```
Comment (lucaswoj) [inline js/symbol/glyph_source.js]: This is exactly how I would start 👌 .

The next step is to have the error event be propagated up to `Style` (from where it'll be propagated up to `Map`). You can see [how we do this on `ImageSprite` here.](http://github.com/mapbox/mapbox-gl-js/blob/abc7d0f2fbbd3beb8d8ec9a9753450b527e5133e/js/style/image_sprite.js#L25-L26)

**Comment by jfirebaugh** [comment]
Hey @stepankuzmin -- I'm in the process of cleaning out our PR queue and it looks like there hasn't been any activity on this one recently. If you're still interested in coming back to it, please let us know and we'll be happy to help you get it to completion. Until then, I'm going to close it out. Thanks again for your contributions!

**Comment by stepankuzmin** [comment]
Hi @jfirebaugh! I'm interested in completing this PR. @lucaswoj said that we need to propagate error event up to `Style`. I've updated the PR in 1e990d62a6a903bf05d2906e2f7470253daf9f61. What should I do next?

**Comment by jfirebaugh** [comment]
Oh, I missed that, sorry! That should be it. Can you rebase on `master` to make sure the build passes?

**Comment by stepankuzmin** [comment]
Whoops. I'm sorry, but it seems that I made `merge` instead of `rebase`. Is there any way to revert it?

**Comment by jfirebaugh** [comment]
Hmm, looks kind of messy. Want to start fresh with a new PR instead?

**Pull request closed without merging**